sonic3k.asm

```diff
@@ -180106,30 +180106,30 @@ loc_84FBC:
 		dbf	d1,loc_84FBC		; Repeat max times
 		moveq	#0,d0
 		move.l	d0,(a0)
-		move.l	d0,$10(a0)
-		move.l	d0,$14(a0)
-		move.b	d0,$2C(a0)
-		move.b	d0,4(a0)
-		move.w	d0,$2A(a0)		; If no open slots, then destroy this object period
+		move.l	d0,x_pos(a0)
+		move.l	d0,y_pos(a0)
+		move.b	d0,subtype(a0)
+		move.b	d0,render_flags(a0)
+		move.w	d0,status(a0)		; If no open slots, then destroy this object period
 		addq.w	#8,sp
 		rts
 ; ---------------------------------------------------------------------------
 
 loc_84FE4:
 		bset	d0,(a2)			; Turn this slot on
-		move.b	d0,$3B(a0)
-		move.w	a2,$3C(a0)		; Keep track of slot address and bit number
-		move.w	d3,$A(a0)		; Use correct VRAM offset
-		move.l	(a1)+,$C(a0)	; Mapping address
-		move.w	(a1)+,8(a0)		; Priority
-		move.b	(a1)+,7(a0)		; Width
-		move.b	(a1)+,6(a0)		; Height
-		move.b	(a1)+,$22(a0)	; Frame number
-		move.b	(a1)+,$28(a0)	; Collision number
-		bset	#2,$2A(a0)		; Turn object slotting on
-		move.b	#-1,$3A(a0)		; CHECKLATER
-		bset	#2,4(a0)		; Use screen coordinates
-		addq.b	#2,5(a0)		; Next routine
+		move.b	d0,ros_bit(a0)
+		move.w	a2,ros_addr(a0)		; Keep track of slot address and bit number
+		move.w	d3,art_tile(a0)		; Use correct VRAM offset
+		move.l	(a1)+,mappings(a0)	; Mapping address
+		move.w	(a1)+,priority(a0)		; Priority
+		move.b	(a1)+,width_pixels(a0)		; Width
+		move.b	(a1)+,height_pixels(a0)		; Height
+		move.b	(a1)+,mapping_frame(a0)	; Frame number
+		move.b	(a1)+,collision_flags(a0)	; Collision number
+		bset	#2,status(a0)		; Turn object slotting on
+		move.b	#-1,objoff_3A(a0)		; Reset objoff_3A (used by Perform_DPLC) 
```

Comment:
Maybe add a constant named `last_dplc_frame` to sonic3k.constants.asm and using it here would be better?